google/cloud/storage/internal/generic_request.h

```diff
@@ -63,11 +63,19 @@ class GenericRequestBase;
 template <typename Derived, typename Option>
 class GenericRequestBase<Derived, Option> {
  public:
+  // Can't use capture by unversal references because if it were a template
+  // guarded by `std::enable_if<>` it would shadow the exact same superclass'
+  // member function.
   Derived& set_option(Option&& p) {
     option_ = std::move(p);
     return *static_cast<Derived*>(this);
   }
 
+  Derived& set_option(Option const& p) {
```

Comment:
Rather than adding an overload, how about just having a single `set_option(Option)` that accepts by value? That would be simpler and it should also fixed the linked issue. Thoughts?